pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/StorageQuotaChecker.java

```diff
@@ -63,12 +63,22 @@ public StorageQuotaChecker(AbstractTableConfig tableConfig, TableSizeReader tabl
   public QuotaCheckerResponse isSegmentStorageWithinQuota(@Nonnull File segmentFile, @Nonnull String tableNameWithType,
       @Nonnull String segmentName,
       @Nonnegative int timeoutMsec) {
-    Preconditions.checkNotNull(segmentFile);
-    Preconditions.checkNotNull(tableNameWithType);
-    Preconditions.checkNotNull(segmentName);
-    Preconditions.checkArgument(timeoutMsec > 0, "Timeout value must be > 0, input: %s", timeoutMsec);
-    Preconditions.checkArgument(segmentFile.exists(), "Segment file: %s does not exist", segmentFile);
-    Preconditions.checkArgument(segmentFile.isDirectory(), "Segment file: %s is not a directory", segmentFile);
+    try {
+      Preconditions.checkNotNull(segmentFile);
```

Comment:
These function params have @Nonnull annotations meaning they don't expect null values to be passed in. So, if a caller passes nulls that is an error. Add error messages to Preconditions if you want but for NPE there is little additional info to report.  These should definitely not be inside try-catch.

Caller should correctly handle these bad values returning appropriate error responses before calling this method.